controllers/articles.controller.js

```diff
@@ -15,8 +16,11 @@ exports.getTopics = (req, res, next) => {
 
 exports.getArticlesById = (req, res, next) => {
   const id = req.params.article_id;
-  fetchArticleById(id)
-    .then((articles) => {
+  Promise.all([fetchArticleById(id), fetchCommentsByArticleId(id)])
+    .then((result) => {
+      const articles = result[0];
+      const comments = result[1];
+      articles[0].comment_count = comments.length;
```

Comment:
I like what you're doing here - but it requires 2 separate queries to the database to achieve. Can this be done with a single query?